### DISCUSSION TITLE
Rethink the branching system

### DISCUSSION THREAD

**Discussion by MichaelDeBoey** [remix-run/remix#1384] [Proposals]
Hi team! 👋

I was thinking about the current branching system that's in place, as I think it has some flaws.

Currently there are coming a lot of PRs that are adding examples, which is awesome.
This however is pointing out a problem in our branching system for me.

As all of these examples are pushed to `main`, `dev` won't have these.
This doesn't necessarily have to be a problem, but could be.

Let's take for example #923, which should go into `dev` according to the policy.
It would be nice that this PR could also change configs for all examples to the new format implemented in that PR.
This however isn't possible, as all these example projects aren't available in `dev`.

I can see more problems arise with having this kind of structure for a repo like we have.

I think we have multiple solutions for this problem:

- Get rid of the multiple branches
  But what about fixing typos in the docs vs adding docs for new (unreleased) features?
- Make sure `dev` is synced with `main`
  This could be done manually, but that would become cumbersome to do each time.
  So I think it would be good to have a script that tries to rebase `dev` onto `main` or merge `main` into `dev` (I personally prefer rebasing).
  If it fails to do so (because of conflicts), we can maybe add an automated issue that's logging the errors it got?

Happy to hear your thoughts about this.

CC/ @kentcdodds @mjackson

**Comment by kentcdodds** [comment]
Thanks for bringing this up. Good points here. I think I prefer the script that keeps dev up-to-date.
